api/src/main/java/org/pmiops/workbench/api/UserAdminController.java

```diff
@@ -0,0 +1,132 @@
+import org.pmiops.workbench.model.UserAuditLogQueryResponse;
+import org.pmiops.workbench.profile.ProfileService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserAdminController implements UserAdminApiDelegate {
+
+  private final UserService userService;
+  private final ProfileService profileService;
+  private final ActionAuditQueryService actionAuditQueryService;
+  private final UserDao userDao;
+  private final Provider<DbUser> userProvider;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  public UserAdminController(
+      UserService userService,
+      ProfileService profileService,
+      ActionAuditQueryService actionAuditQueryService,
+      UserDao userDao,
+      Provider<DbUser> userProvider,
+      Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.userService = userService;
+    this.profileService = profileService;
+    this.actionAuditQueryService = actionAuditQueryService;
+    this.userDao = userDao;
+    this.userProvider = userProvider;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+  }
+
+  @Override
+  @AuthorityRequired({Authority.ACCESS_CONTROL_ADMIN})
+  public ResponseEntity<EmptyResponse> bypassAccessRequirement(
+      Long userId, AccessBypassRequest bypassed) {
+    userService.updateBypassTime(userId, bypassed);
+    return ResponseEntity.ok(new EmptyResponse());
+  }
+
+  @Override
+  @AuthorityRequired({Authority.ACCESS_CONTROL_ADMIN})
+  public ResponseEntity<AdminUserListResponse> getAllUsers() {
+    return ResponseEntity.ok(
+        new AdminUserListResponse().users(profileService.getAdminTableUsers()));
+  }
+
+  @Override
+  @AuthorityRequired({Authority.RESEARCHER_DATA_VIEW})
+  public ResponseEntity<UserAuditLogQueryResponse> getAuditLogEntries(
+      String usernameWithoutGsuiteDomain,
+      Integer limit,
+      Long afterMillis,
+      @Nullable Long beforeMillis) {
+    final String username =
+        String.format(
+            "%s@%s",
+            usernameWithoutGsuiteDomain,
+            workbenchConfigProvider.get().googleDirectoryService.gSuiteDomain);
+    final long userDatabaseId = userService.getByUsernameOrThrow(username).getUserId();
+    final Instant after = Instant.ofEpochMilli(afterMillis);
+    final Instant before =
+        Optional.ofNullable(beforeMillis).map(Instant::ofEpochMilli).orElse(Instant.now());
+    return ResponseEntity.ok(
+        actionAuditQueryService.queryEventsForUser(userDatabaseId, limit, after, before));
+  }
+
+  /**
+   * Gets a JSON list of users and their registered tier access expiration dates.
+   *
+   * <p>This endpoint is intended as a temporary manual measure to assist with user communication
+   * during the rollout of Annual Access Renewal (AAR).
+   *
+   * <p>Once fully rolled out, we will have an automated expiration email process and this can
```

Comment:
Can this be removed now?

Reply:
It appears this isn't used client side - so perhaps this was only being called manually via API. I will remove - let me know if it's still needed

Reply:
Not clear what this is referring to, b/c "Outdated" - going to assume this is getRegisteredTierExpirations()

it's true that this was API only.  Maybe OK to remove - but please check w/ PO about the likelihood of needing this again.

Reply:
confirmed we can remove